spec/unit/http/resolver_spec.rb

```diff
@@ -70,6 +70,23 @@
       expect(service).to be_an_instance_of(Puppet::HTTP::Service::Ca)
       expect(service.url.to_s).to eq("https://apple.example.com:8142/puppet-ca/v1")
     end
+
+    it 'resolves once per session' do
+      failed = stub_request(:get, "https://ca.example.com:8141/status/v1/simple/master").to_return(status: 503)
+      passed = stub_request(:get, "https://apple.example.com:8142/status/v1/simple/master").to_return(status: 200)
+
+      service = subject.resolve(session, :puppet)
+      expect(service.url.to_s).to eq("https://apple.example.com:8142/puppet/v3")
+
+      service = subject.resolve(session, :fileserver)
+      expect(service.url.to_s).to eq("https://apple.example.com:8142/puppet/v3")
+
+      service = subject.resolve(session, :report)
+      expect(service.url.to_s).to eq("https://apple.example.com:8142/puppet/v3")
+
+      expect(failed).to have_been_requested
+      expect(passed).to have_been_requested
```

Comment:
This test doesn't make sense to me. In webmock, if a request happens, does it ensure it only happens once? I see in the new test you just added, you expect the request twice. Here, do we expect each request only once? Same question for down below. If yes, then that does make sense to me, but I wanted to make sure I'm understanding this

Reply:
yep, `expect(..).to have_been_requested` implies the request is only made `once`. For more than once, you can say things like `have_been_requested.twice` or `have_been_requested.times(3)`

Reply:
Oh perfect. I couldn't find anything about that in the webmock documentation at all. Thanks for the confirmation.